### PR TITLE
Origin/bugfix 26

### PR DESCRIPTION
**Pull request by flight-alert**
This is incomplete: before deployment to prod, there must be a migration of old entries in the prod DB that splits full name into first and last.

additionally, search by name is now wonky: it will try to match the first word in the query string to a first name, and the last word to a last name. names such as Aaron Bighead Clemence might not show up if the entire full name is searched.

Some minor tweaks were made that are outside the scope of the issue:
1) name of the filtering button in the volunteer table was changed from לערוך to סינון
2) use of consts in models to set char field lengths

### CODE REVIEW

**Comment by itay-sho** [comment]
I've changed the pull request to be to 'develop' and not 'master', If you did intend to merge directly to master, let's discuss the consequences

File: client/forms.py [hunk 1]
```diff
@@ -54,23 +54,29 @@ class VolunteerForm(forms.Form):
     )
 
     my_validator = RegexValidator(r"^\d+$")
-    full_name = forms.CharField(max_length=200)
-    identity = forms.CharField(max_length=9, validators=[my_validator])
+    #full_name = forms.CharField(max_length=DEFAULT_MAX_FIELD_LENGTH)
```
Comment (itay-sho) [inline client/forms.py]:
```suggestion
```

File: client/forms.py [hunk 2]
```diff
@@ -80,32 +86,36 @@ class VolunteerForm(forms.Form):
 
     def __init__(self, *args, **kwargs):
         super(forms.Form, self).__init__(*args, **kwargs)
-        self.fields['full_name'].label = "שם מלא"
+        #self.fields['full_name'].label = "שם מלא"
```
Comment (itay-sho) [inline client/forms.py]:
```suggestion
```

File: client/forms.py [hunk 3]
```diff
@@ -80,32 +86,36 @@ class VolunteerForm(forms.Form):
...
+        self.fields['neighborhood'].label = "שכונה"
         self.fields['address'].label = "כתובת מגורים"
         self.fields['available_on_saturday'].label = "האם זמין בשבת"
         self.fields['notes'].label = "הערות"
-        self.fields['age'].label = "גיל"
         self.fields['transportation'].label = "דרכי התניידות"
         self.fields['hearing_way'].label = "איך שמעת עלינו"
+
+
+
```
Comment (itay-sho) [inline client/forms.py]:
```suggestion

```

File: client/migrations/0002_auto_20200326_1409.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    operations = [
+        migrations.AddField(
+            model_name='volunteer',
+            name='first_name',
+            field=models.CharField(default='abcdef', max_length=200),
+        ),
+        migrations.AddField(
+            model_name='volunteer',
+            name='last_name',
+            field=models.CharField(default='abcdef', max_length=200),
```
Comment (itay-sho) [inline client/migrations/0002_auto_20200326_1409.py]: maybe make that blank=True field for now? we shouldn't use the abcdef default here

File: client/models.py [hunk 5]
```diff
@@ -63,25 +68,41 @@ class Volunteer(Timestampable):
...
+        ("PROJECT", "פרויקט"),
+        ("CHILD_CARE", "משפחתון"),
+        ("MISSIONS", "משימות")
+    )
 
-    tz_number = models.CharField(max_length=11, blank=True)
-    full_name = models.CharField(max_length=200)
+    tz_number = models.CharField(max_length=ID_LENGTH, blank=True)
+    first_name = models.CharField(max_length=DEFAULT_MAX_FIELD_LENGTH, default="abcdef")
+    last_name = models.CharField(max_length=DEFAULT_MAX_FIELD_LENGTH, default="abcdef")
```
Comment (itay-sho) [inline client/models.py]: `
```suggestion
    first_name = models.CharField(max_length=DEFAULT_MAX_FIELD_LENGTH, blank=True)
    last_name = models.CharField(max_length=DEFAULT_MAX_FIELD_LENGTH, blank=True)
```
consider blank=True without default

File: client/models.py [hunk 6]
```diff
@@ -63,25 +68,41 @@ class Volunteer(Timestampable):
...
+    full_name = models.CharField(max_length=DEFAULT_MAX_FIELD_LENGTH, blank=True)
+    organization = models.CharField(max_length=DEFAULT_MAX_FIELD_LENGTH, blank=True)
     age = models.IntegerField()
+    volunteer_type = models.CharField(max_length=DEFAULT_MAX_FIELD_LENGTH, choices=TYPES, default="MISSIONS")
     areas = models.ManyToManyField(Area)
     languages = models.ManyToManyField(Language)
-    phone_number = models.CharField(max_length=200)
-    email = models.CharField(max_length=200, blank=True)
+    phone_number = models.CharField(max_length=DEFAULT_MAX_FIELD_LENGTH)
```
Comment (itay-sho) [inline client/models.py]: shouldn't phone field be limited to phone number size? (10)
```suggestion
    phone_number = models.CharField(max_length=10)
```

Comment (itay-sho) [reply client/models.py]: you can use constant if you prefer

File: client/forms.py [hunk 7]
```diff
@@ -54,23 +54,29 @@ class VolunteerForm(forms.Form):
...
     age = forms.IntegerField()
-    area = forms.MultipleChoiceField(choices = AREAS, widget=forms.CheckboxSelectMultiple())
+    area = forms.MultipleChoiceField(choices=AREAS, widget=forms.CheckboxSelectMultiple())
     languages = forms.MultipleChoiceField(choices = get_the_lang_choices, widget=forms.CheckboxSelectMultiple())
-    phone_number = forms.CharField(max_length=200)
-    email = forms.CharField(max_length=200)
-    city = forms.ChoiceField(choices = CITIES)
-    address = forms.CharField(max_length=200)
+    phone_number = forms.CharField(max_length=DEFAULT_MAX_FIELD_LENGTH)
+    email = forms.CharField(max_length=DEFAULT_MAX_FIELD_LENGTH)
```
Comment (itay-sho) [inline client/forms.py]: Consider changing email to EmailField

File: client/models.py [hunk 8]
```diff
@@ -63,25 +68,41 @@ class Volunteer(Timestampable):
...
+    email = models.CharField(max_length=DEFAULT_MAX_FIELD_LENGTH, blank=True)
```
Comment (itay-sho) [inline client/models.py]: consider changing email to EmailField

File: client/views.py [hunk 9]
```diff
@@ -36,10 +36,16 @@ def thanks_volunteer(request):
...
         vr = Volunteer.objects.get(pk=vol_id)
-
-        print(vr.full_name)
+        full_name = vr.first_name + " " + vr.last_name
+        # print(vr.full_name)
+        # return render(request, 'thanks_volunteer.html', {
+        #     "name": vr.full_name,
+        #     "taz": vr.tz_number
+        #
+        # })
```
Comment (itay-sho) [inline client/views.py]:
```suggestion
```

File: client/views.py [hunk 10]
```diff
@@ -82,10 +88,13 @@ def volunteer(request):
             keep_mandatory_worker_children = False
             if answer["childrens"] == "YES":
                 keep_mandatory_worker_children = True
-            volunter_new = Volunteer(tz_number=answer["identity"], full_name=answer["full_name"], email=answer["email"],
-                                     age=answer["age"],
+            volunter_new = Volunteer(tz_number=answer["identity"], first_name=answer["first_name"],
+                                     last_name=answer["last_name"], #full_name=answer["full_name"],
```
Comment (itay-sho) [inline client/views.py]:
```suggestion
                                     last_name=answer["last_name"],
```

File: server/views.py [hunk 11]
```diff
@@ -91,7 +91,11 @@ def show_all_volunteers(request, page=1):
 
     if len(search_name) != 0:
         something_mark = True
-        qs = qs.filter(full_name=search_name[0])
+        split_name = search_name[0].split()
+        # qs_first_names = qs.filter(first_name=split_name[0])
+        # qs_last_names = qs.filter(last_name=split_name[:-1])
```
Comment (itay-sho) [inline server/views.py]:
```suggestion
```

File: server/views.py [hunk 12]
```diff
@@ -91,7 +91,11 @@ def show_all_volunteers(request, page=1):
...
+        qs = qs.filter(Q(last_name=split_name[-1]) | Q(first_name=split_name[0]))
+        #qs = qs.filter(full_name=search_name[0])
```
Comment (itay-sho) [inline server/views.py]:
```suggestion
```

File: server/views.py [hunk 13]
```diff
@@ -150,7 +154,7 @@ def show_all_volunteers(request, page=1):
         field = "-" + field
         match_qs = match_qs.order_by(field)
 
-    # ----- check for each volunterr how much times he apper
+    # ----- check for each volunteקr how much times he apper
```
Comment (itay-sho) [inline server/views.py]:
```suggestion
    # ----- check for each volunteer how much times he apper
```

File: client/migrations/0002_auto_20200326_1409.py (hunk 4 above)
Comment (flight-alert) [reply client/migrations/0002_auto_20200326_1409.py]: According to app logic each person must have a first name and a last name, so blank=True seems like it violates our goals. Django does not seem to allow non-blank fields without default values, hence the abcdef...

File: client/models.py (hunk 6 above)
Comment (flight-alert) [reply client/models.py]: I'm not sure why the original max length was 200 - figured maybe they wanted to allow international numbers, organization numbers, or some other unusual cases. Either way I didn't want to change the model without entirely understanding it, I just wanted to get some of those magic numbers out of the code

File: client/forms.py (hunk 7 above)
Comment (flight-alert) [reply client/forms.py]: See above - I didn't really want to change any of the field names and so on. Maybe we can open a seperate issue that's about cleaning things up a bit, there's plenty to go around

File: client/migrations/0002_auto_20200326_1409.py (hunk 4 above)
Comment (itay-sho) [reply client/migrations/0002_auto_20200326_1409.py]: I see, so ignore the blank=True suggestion
but what about setting default to an empty string?

File: client/models.py (hunk 6 above)
Comment (itay-sho) [reply client/models.py]: Gotcha,
can you open issue for that? we might want to handle this on the future

File: client/migrations/0002_auto_20200326_1409.py (hunk 4 above)
Comment (flight-alert) [reply client/migrations/0002_auto_20200326_1409.py]: Could work better, I suppose. I'll make the change.

**Pull request merged**